subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -164,11 +162,12 @@ static void send_pending_adv(struct k_work *work)
 			continue;
 		}
 
-		BT_MESH_ADV(buf)->busy = 0U;
 		err = buf_send(buf);
 		if (!err) {
 			return; /* Wait for advertising to finish */
```

Comment:
Aren't you leaking `adv.buf` here? In your modifications to `buf_send()` it remains set both for success and failure.

Reply:
At present, it seems that this set is not very important. It will be emptied when it is reassigned, but it can be added again.